jaxrs-client/src/main/java/io/micronaut/jaxrs/client/JaxRsClientBuilder.java

```diff
@@ -0,0 +1,189 @@
+        jaxRsConfiguration.register(httpClient.getHandlerRegistry().findReader(Argument.of(byte[].class), List.of(MediaType.ALL_TYPE)).get());
+        jaxRsConfiguration.register(httpClient.getHandlerRegistry().findWriter(Argument.STRING, List.of(MediaType.ALL_TYPE)).get());
+        jaxRsConfiguration.register(httpClient.getHandlerRegistry().findWriter(Argument.of(byte[].class), List.of(MediaType.ALL_TYPE)).get());
+        jaxRsConfiguration.register(new JaxRsReaderMessageBodyWriter());
+        jaxRsConfiguration.register(new JaxRsReaderMessageBodyWriter());
+        jaxRsConfiguration.register(new JaxRsReaderMessageBodyReader());
+        jaxRsConfiguration.register(new JaxRsInputStreamMessageBodyWriter<>());
+        jaxRsConfiguration.register(new JaxRsInputStreamMessageBodyReader());
+        jaxRsConfiguration.register(new JaxRsStreamingOutputMessageBodyWriter<>());
+
+        if (TESTING_MIN_CLIENTS > 0) {
+            TESTING_CLIENTS.removeIf(w -> w.get() == null);
+            TESTING_CLIENTS.add(new WeakReference<>(httpClient));
+            if (TESTING_CLIENTS.size() > TESTING_MIN_CLIENTS) {
+                DefaultHttpClient client = TESTING_CLIENTS.remove(0).get();
+                if (client != null) {
+                    client.close();
+                }
+            }
+        }
+        return new JaxRsClient(httpClient, jaxRsConfiguration);
+    }
+
+    @Override
+    public ClientBuilder self() {
+        return this;
+    }
+
+    @Override
+    public JaxRsConfiguration getConfiguration() {
+        return config;
+    }
+
+    @Override
+    public ClientBuilder withConfig(Configuration config) {
+//        this.config = config;
+        return this;
+    }
+
+    @Override
+    public ClientBuilder sslContext(SSLContext sslContext) {
+        this.sslContext = sslContext;
+        return this;
+    }
+
+    @Override
+    public ClientBuilder keyStore(KeyStore keyStore, char[] password) {
+        keyStores.put(keyStore, password);
+        return this;
+    }
+
+    @Override
+    public ClientBuilder trustStore(KeyStore trustStore) {
+        this.trustStore = trustStore;
+        return this;
+    }
+
+    @Override
+    public ClientBuilder hostnameVerifier(HostnameVerifier verifier) {
+        this.hostnameVerifier = verifier;
+        return this;
+    }
+
+    @Override
+    public ClientBuilder executorService(ExecutorService executorService) {
+        this.executorService = executorService;
+        return this;
+    }
+
+    @Override
+    public ClientBuilder scheduledExecutorService(ScheduledExecutorService scheduledExecutorService) {
+        this.scheduledExecutorService = scheduledExecutorService;
+        return this;
+    }
+
+    @Override
+    public ClientBuilder connectTimeout(long timeout, TimeUnit unit) {
+        this.connectTimeout = Duration.ofMillis(unit.toMillis(timeout));
```

Comment:
```suggestion
        this.connectTimeout = Duration.of(timeout, unit.toChronoUnit());
```

A bit cleaner so that µs/ns arent lost